core/src/main/java/com/github/jsonldjava/utils/JsonUtils.java

```diff
@@ -371,6 +363,56 @@ public static Object fromURL(java.net.URL url, CloseableHttpClient httpClient)
         }
     }
 
+    private static InputStream getJsonLdViaHttpUri(final URL url, final CloseableHttpClient httpClient,
+            CloseableHttpResponse response) throws IOException {
+        final HttpUriRequest request = new HttpGet(url.toExternalForm());
+        // We prefer application/ld+json, but fallback to application/json
+        // or whatever is available
+        request.addHeader("Accept", ACCEPT_HEADER);
+        response = httpClient.execute(request);
+
+        final int status = response.getStatusLine().getStatusCode();
+        if (status != 200 && status != 203) {
+            throw new IOException("Can't retrieve " + url + ", status code: " + status);
+        }
+        // follow alternate document location
+        // https://www.w3.org/TR/json-ld11/#alternate-document-location
+        URL alternateLink = alternateLink(url, response);
+        if (alternateLink != null) {
+            return getJsonLdViaHttpUri(alternateLink, httpClient, response);
```

Comment:
This recursion may not complete if the document or any of its downstream links resolves to a previous link. Either a set of the previous links needs to be maintained or this needs to be converted to not rely on recursion.

Reply:
Treated with 1d675d7e1e4ccde0545b336d98342439bc73da4e: a (sowmehow arbitrarily) recursion of depth 20 (MAX_LINKS_FOLLOW = 20) is allowed. (As a sidenote: I see your point - it's always good to be robust. But I would like to say, if such a case occurs, that the data (the link header) is erroneous and should be fixed by the provider of the data).

Reply:
It is great to get providers to fix their setups, but this would cause client-side issues if it were not securely mitigated here so the hardcoded follow limit is still very useful.

Just a comment on the resolution to this. Would prefer it throws `IOException` rather than return null to use the existing exception handlers for this case.